docs/code-style.md

```diff
@@ -1,6 +1,237 @@
 # Code Style
 
-The [pom.xml](pom.xml) is configured to enforce a coding style defined in [checkstyle.xml](checkstyle.xml).
+The intent here is to maintain a common style across the project and rely on the process to enforce it instead of
+individuals.
 
-The intent here is to maintain a common style across the project 
-and rely on the process to enforce it instead of individuals.
+## Automated code style check
+
+The [pom.xml](../pom.xml) is configured to enforce a coding style defined in [checkstyle.xml](../checkstyle.xml) when
+maven `validate` phase executed.
+
+## Formatting
+
+The project uses formatting rules described in `.editorconfig` file. Most of the popular IDEs have support of it. For
+example, in `IntelliJ IDEA` hit `Code -> Reformat Code` to organize your code.
+
+## Code conventions
```

Comment:
I'd also add some general and simple things like:
- Prefer composition over inheritance.
- Methods should have only 1 return point (not enforced)

Reply:
i'd say it's more like design. let me skip them.